### PR TITLE
Unstructured cleanups

### PR DESCRIPTION
**Pull request by ash2k**
**What this PR does / why we need it**:
Cleanups for `Unstructured`/`UnstructuredList` extracted from #55297.

**Release note**:
```release-note
NONE
```
/sig api-machinery
/kind enhancement
/assign @sttts

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go [hunk 1]
```diff
@@ -162,7 +161,12 @@ func (u *Unstructured) GetOwnerReferences() []metav1.OwnerReference {
 func (u *Unstructured) SetOwnerReferences(references []metav1.OwnerReference) {
 	newReferences := make([]interface{}, 0, len(references))
 	for _, reference := range references {
-		newReferences = append(newReferences, setOwnerReference(reference))
+		out, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&reference)
```
Comment (ash2k) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go]: This is consistent with how method `SetInitializers()` (line 357) works, future-proof against new fields and takes into account all json tags properly (see bug with pointers #46817).

**Comment by caesarxuchao** [comment]
cc @cheftako

**Pull request merged**